c2corg_ui/static/js/documentservice.js

```diff
@@ -134,16 +141,21 @@ app.Document.prototype.pushToAssociations = function(doc, doctype,
 /**
  * @param {number} id Id of document to unassociate
  * @param {string} type Type of document to unassociate
+ * @param {goog.events.Event | jQuery.Event} [event]
  * @export
  */
-app.Document.prototype.removeAssociation = function(id, type) {
+app.Document.prototype.removeAssociation = function(id, type, event) {
+  event.currentTarget.closest('.list-item').className += ' remove-item';
   var associations = this.document.associations[type];
-  for (var i = 0; i < associations.length; i++) {
-    if (associations[i]['document_id'] === id) {
-      associations.splice(i, 1);
-      break;
+  setTimeout(function () {
```

Comment:
if you remove it instantly, it will disappear instantly because of two-way angular binding, which will automatically remove the object from ng-repeat.